src/fides/api/ctl/routes/system.py

```diff
@@ -219,21 +227,48 @@ async def upsert(
     }
 
 
+def privacy_declaration_logical_id(
+    privacy_declaration: PrivacyDeclaration,
+) -> str:
+    """
+    Helper to standardize a logical 'id' for privacy declarations.
+    As of now, this is based on the `data_use` and the `name` of the declaration, if provided.
+    """
+    return f"{privacy_declaration.data_use}:{privacy_declaration.name or ''}"
+
+
 async def upsert_privacy_declarations(
     db: AsyncSession, resource: SystemSchema, system: System
 ) -> None:
     """Helper to handle the specific upsert logic for privacy declarations"""
 
     async with db.begin():
-        # clear out any existing privacy declarations on the System
-        for existing_declaration in system.privacy_declarations:
-            await db.delete(existing_declaration)
-
-        # iterate through declarations specified on the request and create them
+        # map existing declarations by their logical identifier
+        existing_declarations: Dict[str, PrivacyDeclaration] = {
+            privacy_declaration_logical_id(existing_declaration): existing_declaration
+            for existing_declaration in system.privacy_declarations
+        }
+
+        # iterate through declarations specified on the request and upsert
+        # looking for "matching" existing declarations based on data_use and name
         for privacy_declaration in resource.privacy_declarations:
+            # prepare our 'payload' for either create or update
             data = privacy_declaration.dict()
-            data["system_id"] = system.id  # add FK back to system
-            PrivacyDeclaration.create(db, data=data)
+            data["system_id"] = system.id  # include FK back to system
+
+            # if we find matching declaration, remove it from our map
+            if existing_declaration := existing_declarations.pop(
+                privacy_declaration_logical_id(privacy_declaration), None
+            ):
+                # and update existing declaration *in place*
+                existing_declaration.update(db, data=data)
+            else:
+                # otherwise, create a new declaration record
+                PrivacyDeclaration.create(db, data=data)
```

Comment:
oh interesting in testing with when adjusting name or data use, I'm usually seeing a new one get created and the old one get deleted which makes sense in hindsight. understood though that we're blocking these edits from the UI

Reply:
yup, good observation. but that's exactly why we want to block those edits from the UI!

agreed that it's generally a bit fraught. goes back to the broader discussion about how we probably want to move more toward an imperative API squared directly on `PrivacyDeclaration`s (or maybe they will be renamed) in the medium term future.